plugins/varnish/varnish5_

```diff
@@ -0,0 +1,1177 @@
+	
+	if ($varnishname) {
+		$arg .= " -n $varnishname";
+	}
+
+	open (XMLDATA, "$varnishstatexec $arg|") or die "meh";
+	$parser->parse(*XMLDATA, ProtocolEncoding => 'ISO-8859-1');
+	close(XMLDATA);
+}
+
+# Prints the fields in the list in $_[2] (e.g: 'value'/'description') for
+# each identity of the varnish counter/family combination as defined by
+# the $_[0]-counter on the aspect definition. Err, that's jibberish, so
+# an example:
+#
+# e.g: dynamic_print('SMA_1','',('value'))
+# e.g: dynamic_print('SMA_2','.label',('ident','description'))
+# SMA_1 is the counter-value. If it is a dynamic counter, it has a counter
+# and family-member (e.g: counter: c_req, family: SMA) and print_dynamic
+# will print c_req for each SMA-identity.
+#
+# Note that the variables to print is a list. This is to allow printing a
+# single item with multiple fields. Typically for identity+description so
+# you can distinguish between different data points.
+#
+# Returns true if it was a dynamic counter.
+sub print_dynamic
+{
+	my $name = $_[0];
+	shift;
+	my $suffix = $_[0];
+	shift;
+	my @field = @_;
+	if (!defined($ASPECTS{$self}{'values'}{$name}{'counter'})) {
+		return 0;
+	}
+	if (!defined($ASPECTS{$self}{'values'}{$name}{'family'})) {
+		return 0;
+	}
+	my $counter = $ASPECTS{$self}{'values'}{$name}{'counter'};
+	my $type = $ASPECTS{$self}{'values'}{$name}{'family'};
+	
+	foreach my $key (keys %{$data{$type}}) {
+		my $pname = normalize_name($type . "_" . $key . "_" . $counter);
+		print $pname . $suffix . " ";
+		my $i = 0;
+		foreach my $f (@field) {
+			if ($i != 0) {
+				print " ";
+			}
+			$i += 1;
+			print $data{$type}{$key}{$counter}{$f};
+		}
+		print "\n";
+	}
+	return 1;
+}
+
+# Read and verify the aspect ($self).
+sub set_aspect
+{
+	$self = $0;
+	$self =~ s/^.*\/varnish[0-9]?_//;
+	if (!defined($ASPECTS{$self}) && @ARGV == 0) {
+		usage "No such aspect";
+	}
+}
+
+# Print 'yes' if it's reasonable to use this plugin, or 'no' with a
+# human-readable error message. Always exit true, even if the response
+# is 'no'.
+sub autoconf
+{
+	# XXX: Solaris outputs errors to stderr and always returns true.
+	# XXX: See #873
+	if (`which $varnishstatexec 2>/dev/null` =~ m{^/}) {
+		print "yes\n";
+	} else {
+		print "no ($varnishstatexec could not be found)\n";
+	}
+	exit 0;
+}
+
+# Suggest relevant aspects/values of $self.
+# 'DEBUG'-graphs are excluded.
+sub suggest
+{
+	foreach my $key (keys %ASPECTS) {
+		if (defined($ASPECTS{$key}{'DEBUG'}) && $FULL_SUGGEST != 1) {
+			next;
+		}
+		print "$key\n";
+	}
+}
+
+# Walk through the relevant aspect and print all top-level configuration
+# values and value-definitions.
+sub get_config
+{
+	my $graph = $_[0];
+
+	# Need to double-check since set_aspect only checks this if there
+	# is no argument (suggest/autoconf doesn't require a valid aspect)
+	if (!defined($ASPECTS{$graph})) {
+		usage "No such aspect";
+	}
+	my %values = %{$ASPECTS{$graph}{'values'}};
+
+	print "graph_category webserver\n";
```

Comment:
Why this became `webserver` instead of `varnish`? It's kinda confusing

Reply:
It is recommended to use categories from the [list of well-known plugin categories](http://guide.munin-monitoring.org/en/latest/reference/graph-category.html#well-known-categories).

But maybe `loadbalancer` would be more suitable? I cannot tell.